internal/decoder/decoder.go

```diff
@@ -25,12 +26,25 @@ func DecoderForModule(ctx context.Context, mod module.Module) (*decoder.Decoder,
 		d.SetUtmMedium(clientName)
 	}
 
-	for name, f := range mod.ParsedFiles {
+	err := loadFiles(d, mod.ParsedModuleFiles)
+	if err != nil {
+		return nil, err
+	}
+
+	err = loadFiles(d, mod.ParsedVarsFiles)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I know this may looks like a minor thing, but I'd rather see a separate function `DecoderForVariables` here, in a similar vein as `schemaForDocument`. 

I'm not sure what exact implications does it have (probably nothing visible to the user), but I am fairly sure that the decoder does not need both, so it's better to just not load it IMO. Relatedly the decoder for tfvars doesn't need any of the setup above (reference reader, and UTM stuff).

Reply:
done!